Remove more code based on AP_VIDEOTX_ENABLED

peterbarker opened the pull request:
I've run this through the script which disables all build options in turn; the tramp dependency seems to work.

```
Board              AP_Periph  blimp  bootloader  copter  heli  plane  rover  sub
Durandal                      0      0           0       0     0      0      0
Hitec-Airspeed     0                                                         
KakuteH7-bdshot               0      0           0       0     0      0      0
MatekF405                     0      0           0       0     0      0      0
Pixhawk1-1M                   0      0           0       0     0      0      0
f103-QiotekPeriph  0                                                         
f303-Universal     0                                                         
revo-mini                     0      0           0       0     0      0      0
skyviper-v2450                                   -24                         
```

tridge commented on libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp:
```diff
@@ -829,6 +840,7 @@ void AP_CRSF_Telem::update_vtx_params()
         _telem_size = len + 1;
     }
 }
+#endif
```
add end comment //

peterbarker replied:
Done

tridge commented on libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp:
```diff
@@ -600,6 +609,7 @@ void AP_CRSF_Telem::process_vtx_telem_frame(VTXTelemetryFrame* vtx)
 
     _vtx_power_change_pending = _vtx_freq_change_pending = _vtx_options_change_pending = false;
 }
+#endif
```
add end //

andyp1per commented on libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp:
```diff
@@ -389,7 +389,11 @@ bool AP_CRSF_Telem::is_packet_ready(uint8_t idx, bool queue_empty)
     case PARAMETERS:
         return _pending_request.frame_type > 0;
     case VTX_PARAMETERS:
+#if AP_VIDEOTX_ENABLED
```
I'm not really convinced this is a useful thing to do. This assumes you for some reason want to build CRSF telemetry without VTX support. A more useful thing to do would be to make yaapu telemetry optional since that is much bigger flash-wise - but no-one wants to do this anyway. I don't think you will want this for AP_Periph and it just ugifies the code.

peterbarker replied:
Why is building with CRSF telemetry without VTX at all weird?  Neither of my Crossifre-using vehicles has a video transmitter on them.  And I know others in the same boat.

We'd be adding unnecessary coupling between CRSF and the VTX library if we were to make that technical dependency of one on the other.

I didn't actually do it for my daily flyer, but I did just check and that board's surprisingly close to the edge.  Excising videotx removes an additional 10% free space on that board (~+3kB).  Given that both tramp and smartaudio are minimize-features, it seems saving an additional 3kB on minimized boards by adding AP_VIDEOTX_ENABLED is a no-brainer - but I'd like to have the option of CRSF telem on there!

Hwurzburg replied:
I use CRSF....and both HD and analog video..But not smart audio video..on all my vehicles

Hwurzburg replied:
I would be happy to option out FRSKY D and only have passthru as standard....its obsolete

peterbarker replied:
> I would be happy to option out FRSKY D and only have passthru as standard....its obsolete

We already have options to do that one :-)

peterbarker replied:
> @peterbarker the reason I am reluctant is its more testing for me on CRSF. I am not objecting to making video tx optional - just objecting to the small slicing within the CRSF library itself. How much does this actually save, much less than 3k I 
imagine.

It saves an *additional* 3916 bytes on this branch.  That's on top of the ~4500 bytes saved from not compiling in tramp and smartaudio.

```
 everything: bin/arducopter   1813892      3500   258876               1817392          148672
no backends: bin/arducopter   1809388      3500   258876               1812888          153176
 no videotx: bin/arducopter   1805472      3368   259000               1808840          157228
```

>  Do you use yaapu? If you don't there is a mountain of flash to be saved there.

Yes, I use yaapu telemetry.  As does @Hwurzburg , pretty sure.  But this isn't an and/or thing - we can do both.  In fact, I think the bulk of yaapu's stuff might be behind `AP_FRSKY_SPORT_PASSTHROUGH_ENABLED`

Hwurzburg replied:
> > I would be happy to option out FRSKY D and only have passthru as standard....its obsolete
> 
> We already have options to do that one :-)

what I meant was that it could be excluded from the standard builds now with only passthru used

andyp1per reviewed (changes requested):
Don't think the CRSF change is useful

peterbarker commented:
I've removed the agreed-upon MergeOnCIPass and replaced with DevCallEU so we can discuss.  I can remove the changes to crossfire-telemetry and make videotx a hard requirement if you want crossfire - but I'd rather not.

Hwurzburg commented:
@peterbarker extract features needed here too?
